SpringProject/src/main/java/com/gradproject2019/users/service/UserServiceImpl.java

```diff
@@ -15,22 +18,62 @@
 
     private final UserRepository userRepository;
 
+    public static final String PASSWORD_PATTERN = "((?=.*[a-z])(?=.*[0-9])(?=.*[!?\\#@^&£$*+;:~])(?=.*[A-Z]).{8,16})";
+    public static final String EMAIL_PATTERN = "^[a-zA-Z0-9\\.\\!\\#\\$\\%\\&\\'\\*\\+\\-\\/\\=\\?\\^\\_\\`]+@[a-zA-Z0-9]+\\.[\\.A-Za-z]{1,10}";
+    public static final String USERNAME_PATTERN = "^[a-zA-Z0-9]*$";
```

Comment:
Would be nice to have comments on these regexs explaining what they're doing/how they're doing it - if the requirements in the future change for valid usernames/emails/passwords then this wouldn't be very maintainable!

Something that might help: https://alexwlchan.net/2016/04/regexes-are-code/

Reply:
Comments aren't needed, tests explain the implementation

Reply:
Brief comments added into the code and the ticket updated in the board